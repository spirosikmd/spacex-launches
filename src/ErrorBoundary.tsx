import React, { PureComponent } from 'react';
import CssBaseline from '@material-ui/core/CssBaseline';
import { Typography } from '@material-ui/core';

class ErrorBoundary extends PureComponent {
  state = {
    hasError: false,
  };

  componentDidCatch(/* error, info */) {
    this.setState({ hasError: true });
  }

  render() {
    if (this.state.hasError) {
      return (
        <>
          <CssBaseline />
          <Typography>
            Something unexpected happened! Refresh the page!
          </Typography>
        </>
      );
    }

    return this.props.children;
  }
}

export default ErrorBoundary;
