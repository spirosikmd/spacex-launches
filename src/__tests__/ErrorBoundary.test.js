import React from 'react';
import ErrorBoundary from '../ErrorBoundary';

describe('ErrorBoundary', () => {
  let props;

  beforeEach(() => {
    props = {
      children: <div>Test</div>,
    };
  });

  describe('when there is no error', () => {
    it('renders children', () => {
      expect(shallow(ErrorBoundary, props)).toMatchSnapshot();
    });
  });

  describe('when there is error', () => {
    it('renders an error message', () => {
      const errorBoundary = shallow(ErrorBoundary, props);
      errorBoundary.setState({ hasError: true });
      expect(errorBoundary).toMatchSnapshot();
    });
  });
});
