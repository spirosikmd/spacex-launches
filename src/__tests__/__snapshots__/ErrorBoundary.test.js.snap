// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorBoundary when there is error renders an error message 1`] = `
<Fragment>
  <WithStyles(CssBaseline) />
  <WithStyles(Typography)>
    Something unexpected happened! Refresh the page!
  </WithStyles(Typography)>
</Fragment>
`;

exports[`ErrorBoundary when there is no error renders children 1`] = `
<div>
  Test
</div>
`;
